cpp/src/arrow/compute/registry.cc

```diff
@@ -45,6 +45,15 @@ class FunctionRegistry::FunctionRegistryImpl {
     return Status::OK();
   }
 
+  Status AddAlias(const std::string& target_name, const std::string& source_name) {
+    auto it = name_to_function_.find(source_name);
+    if (it == name_to_function_.end()) {
+      return Status::KeyError("No function registered with name: ", source_name);
+    }
+    name_to_function_[target_name] = it->second;
```

Comment:
This is missing synchronization (should be a rare event, but we should be diligent about these issues) -- can you open a JIRA about handling this?